File: C/P003/PrimeNumber.c
#include <stdio.h>
#include <math.h>
void main() {
	int p = 2, n = 3;
	while (1) {
		int m = (int)sqrt(n);
		for (int i = 2; i <= m; i++) {
			int r = n % i;
			if (r == 0)break;
			if (i == m)p = n;
		}
		n++;
		if (n >= 100)break;
	}
	printf("%.2d\n", p);
	system("PAUSE");
}

File: JAVA/P003/src/PrimeNumber.java

public class PrimeNumber {

	public static void main(String[] args) {
		// TODO 1부터 100 사이에서 가장 큰 소수를 구하는 알고리즘을 제시하라
		int p=2,n=3;
		while(true) {
			double tmp = Math.sqrt(n);
			int m =(int)tmp;
			for(int i=2;i<=m;i++) {
				int r=n%i;
				if(r==0) break;
				if(i==m) p=n;
			}
			n++;
			if(n>100) break;
		}
		System.out.println(p);
	}

}
